### DISCUSSION TITLE
J4 IO_IN

### DISCUSSION THREAD

**Discussion by wocard2** [ataradov/usb-sniffer#15] [General]
What exactly is J4 for?

I guess a switch goes there and into the hole in the case, used for triggering transmission of a status packet. The hole is empty in the photograph, perhaps the switch turned out to be unnecessary...

Which switch fits the case?

**Comment by ataradov** [comment]
This is a trigger input. It is just a jumper that fits any standard 0.1" connector (2 singles or one double). The other end of the wires may be connected to the device under test and it can signal when to start and stop the capture.

USB generates a lot of traffic and when debugging an issue that happens at a specific moment in time, sometimes it is easier to just capture stuff that happens at that moment.

For triggering you get an option to capture on rising or falling edge and low or high level.

This is rarely needed, but I had a board space, so I added it just in case.

**Comment by ataradov** [comment]
Note that even if you don't use it as a trigger, the transition on that pin shows in the log. So you can use it as a marker of something happening in the firmware of a debugged device. This may help correlate what the firmware is doing with what happens on the bus.

**Comment by wocard2** [comment]
Thanks! And the hole in the case is for the wires connected to J4 and J5?

**Comment by ataradov** [comment]
Correct. It is sized to fit a cable like this https://www.adafruit.com/product/4934

**Comment by wocard2** [comment]
Thanks!

**Comment by ataradov** [comment]
By default the input is pulled-up, so even passively shorting the pins would show up in the log.
